interface/controller/pkmn_controller.go

```diff
@@ -66,3 +77,34 @@ func (pc *pokemonController) GetPokemonById(c icontext.IContext) error {
 
 	return c.JSON(http.StatusOK, pkmn)
 }
+
+func (pc *pokemonController) GetPokemonConcurrently(c icontext.IContext) error {
+	var p []*model.Pokemon
+
+	items, err := strconv.ParseInt(c.QueryParam(itemsParam), 10, 64)
+	if err != nil || items < 1 {
+		return c.JSON(http.StatusBadRequest, u.ResponseBuilder(http.StatusBadRequest, "query parameter 'items' must be numeric and greater than 0"))
+	}
+
+	itemsWorker, err := strconv.ParseInt(c.QueryParam(itemsPerWorkerParam), 10, 64)
+	if err != nil || itemsWorker == 0 {
+		return c.JSON(http.StatusBadRequest, u.ResponseBuilder(http.StatusBadRequest, "query parameter 'items_per_worker' must be numeric and greater than 0"))
+	}
+
+	if itemsWorker > items {
+		return c.JSON(http.StatusBadRequest, u.ResponseBuilder(http.StatusBadRequest, "'items_per_worker' parameter shouldn't have a greater value than 'items'"))
+	}
+
+	t := strings.ToLower(c.QueryParam(typeParam))
+	if t != "" && strings.Compare(t, odd) != 0 && strings.Compare(t, even) != 0 {
```

Comment:
I guess there's nothing wrong with doing this, but I don't think it's completely necessary to use the Compare function, just a regular `string1 != string2` will work fine